File: app/src/main/java/io/github/getsmp/lorforandroid/ui/section/forum/ForumOverviewItemFactory.java
/*
 * Copyright 2016 getsmp
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.github.getsmp.lorforandroid.ui.section.forum;

import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.ArrayList;
import java.util.List;

import io.github.getsmp.lorforandroid.ui.section.ItemFactory;

public class ForumOverviewItemFactory implements ItemFactory {
    @Override
    public List prepareItems(Element body) {
        List<ForumOverviewItem> items = new ArrayList<ForumOverviewItem>();
        Elements sections = body.select("div#bd").select("ul").first().select("li");
        for (Element section : sections) {
            items.add(new ForumOverviewItem(
                    section.select("a").first().attr("href").replace("/forum/", "").replace("/", ""),
                    section.select("a").first().ownText()
            ));
        }
        return items;
    }
}

File: app/src/main/java/io/github/getsmp/lorforandroid/ui/section/forum/ForumOverviewItemFactory.kt
/*
 * Copyright (c) 2016 getsmp
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package io.github.getsmp.lorforandroid.ui.section.forum

import io.github.getsmp.lorforandroid.ui.section.ItemFactory
import org.jsoup.nodes.Element

class ForumOverviewItemFactory : ItemFactory {
    override fun prepareItems(body: Element, items: MutableList<Any>) {
        val sections = body.select("div#bd").select("ul").first().select("li")
        for (section in sections) {
            items.add(ForumOverviewItem(
                    url = section.select("a").first().attr("href").replace("/forum/", "").replace("/", ""),
                    name = section.select("a").first().ownText()))
        }
    }
}
